sentinelhub/api/catalog.py

```diff
@@ -146,35 +162,68 @@ def search(
 
     @staticmethod
     def _parse_collection_id(collection: Union[str, DataCollection]) -> str:
-        """Extracts catalog collection id from an object defining a collection"""
+        """Extracts catalog collection id from an object defining a collection."""
         if isinstance(collection, DataCollection):
             return collection.catalog_id
         if isinstance(collection, str):
             return collection
         raise ValueError(f"Expected either a DataCollection object or a collection id string, got {collection}")
 
+    def _prepare_filters(
+        self,
+        filter_query: Union[None, str, JsonDict],
+        collection: Union[DataCollection, str],
+        filter_lang: Literal["cql2-text", "cql2-json"],
+    ) -> Union[None, str, JsonDict]:
+        """Asserts that the input coincides with the selected filter language and adds any collection filters."""
+        input_missmatch_msg = f"Filter query is {filter_query} but the filter language is set to {filter_lang}."
+
+        collection_filters = self._get_data_collection_filters(collection)
+
+        if filter_lang == "cql2-text":
+            if not (filter_query is None or isinstance(filter_query, str)):
+                raise ValueError(input_missmatch_msg)
+
+            text_queries = [f"{field}='{value}'" for field, value in collection_filters.items()]
+            if filter_query:
+                text_queries.append(filter_query)
+            return " AND ".join(text_queries) if text_queries else None
+
+        # filter_lang == cql2-json
```

Comment:
what happens if something else gets passed by the user as `filter_lang`, if I understand correctly, then this would be executed as `cql2-json`? 

If I understand this correctly, than either some assert is needed to check that the value is one of these,  or this needs to be rewritten as something like: 

```
if filter_lang == "cql2-text"
  ...
elif filter_lang == "cql2-json"
  ...
else:
  raise ValueError("Unsupported filter-lang")

```

Reply:
I think its best to assume that the user doesn't have type checker and has fat fingers 🤣